webpack/assets/javascripts/react_app/components/SearchBar/index.js

```diff
@@ -1,18 +1,27 @@
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
+import uuid from 'uuid';
 import * as actions from '../AutoComplete/AutoCompleteActions';
 import reducer from '../AutoComplete/AutoCompleteReducer';
 import SearchBar from './SearchBar';
 
+const reducerID = uuid();
+
 const mapStateToProps = ({
   autocomplete: {
-    error, results, searchQuery, status,
-  },
+    [reducerID]: {
```

Comment:
selector meant exactly for this case,  selects the specific record from the store, and speed up the process. 
@sharvit, any thought?

Reply:
IMO the main autocomplete should have a selector like:
```js
selectAutocompleteById = (state, autocompleteId) => state.autocomplete[autocompleteId];
```

Then in each place want to use the autocomplete will need to consume the logic of how to select the data from the state rather than implementing it by himself.
Even it's a simple logic, it's still a logic we will want to reuse and test.

As for the uuid, I think it should be define inside `SearchBarConstants.js` so we can consume the key in different places and get this data into different components.

Reply:
thanks, using selectors now.